ci/cirrus.sh

```diff
@@ -4,6 +4,10 @@ set -eux
 
 export LC_ALL=C
 
+if [ "${GITHUB_ACTIONS:-undefined}" = "true" ]; then
+    git config --global --add safe.directory "$GITHUB_WORKSPACE"
+fi
+
```

Comment:
I think it will be good to do this in the workflow, and keep this file agnostic of the CI. (Okay, it's currently called `cirrus.sh`, and we should anyway change this. ^^)

Reply:
Thanks! [Resolved](https://github.com/bitcoin-core/secp256k1/pull/1396#issuecomment-1668502527).